### PR TITLE
Add max completion time to taskInfo (#881)

### PR DESCRIPTION
**Pull request by edi33416**

### CODE REVIEW

**Comment by edi33416** [comment]
This cherry-picks #881 onto stable, in order to provide `maxCompletionTime` to version 2.x

@gkleiman can you please review?

**Comment by edi33416** [comment]
Please note that I had to remove the `concurrent` field, as it wasn't present on `stable`.
See errors
```
[ERROR] /home/fawkes/chronos/src/main/scala/org/apache/mesos/chronos/utils/JobSerializer.scala:130: error: value concurrent is not a member of org.apache.mesos.chronos.scheduler.jobs.BaseJob
[ERROR]     json.writeBoolean(baseJob.concurrent)
```

```
[ERROR] /home/fawkes/chronos/src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Jobs.scala:127: error: value concurrent overrides nothing
[ERROR]                              @JsonProperty override val concurrent: Boolean = false,
[ERROR]                                                         ^                                                                                                                                                  
[ERROR] /home/fawkes/chronos/src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Jobs.scala:166: error: value concurrent overrides nothing
[ERROR]                                @JsonProperty override val concurrent: Boolean = false,
[ERROR]                                                           ^  
```

File: src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala [hunk 1]
```diff
@@ -127,6 +128,13 @@ class MesosTaskBuilder @Inject()(val conf: SchedulerConfiguration) {
       }
     }
 
+    // Add maxCompletionTime
+    if (job.maxCompletionTime > 0L) {
+      val maxCompletionTime = DurationInfo.newBuilder().setNanoseconds(job.maxCompletionTime * pow(10, 9).toLong).build()
```
Comment (gkleiman) [inline src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala]: I should have noticed this on the previous PR, but better late than never =).

I think we should use `scala.concurrent.duration.Duration` here, like this:

```
val maxCompletionTime = DurationInfo.newBuilder().setNanoseconds(Duration(job.maxCompletionTime.seconds.toNanos).build()
```

File: src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala [hunk 2]
```diff
@@ -127,6 +128,13 @@ class MesosTaskBuilder @Inject()(val conf: SchedulerConfiguration) {
...
+      taskInfo.setMaxCompletionTime(maxCompletionTime)
+      taskInfo.setMaxCompletionTime(maxCompletionTime)
```
Comment (gkleiman) [inline src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala]: Remove this line.

**Review by gkleiman — CHANGES_REQUESTED** [review]
Thanks! Just two minor comments =).

File: src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala (hunk 2 above)
Comment (edi33416) [reply src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala]: Oops. Will remove

File: src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala [hunk 3]
```diff
@@ -127,6 +129,14 @@ class MesosTaskBuilder @Inject()(val conf: SchedulerConfiguration) {
       }
     }
 
+    // Add maxCompletionTime
+    if (job.maxCompletionTime > 0L) {
+      val maxCompletionTime = DurationInfo.newBuilder()
+            .setNanoseconds(job.maxCompletionTime.seconds.toNanos)
```
Comment (edi33416) [inline src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala:135]: @gkleiman is this ok, or would you prefer
`.setNanoseconds(Duration(job.maxCompletionTime, SECONDS).toNanos)` ?

**Comment by gkleiman** [comment]
LGTM, thank you so much!

**Pull request merged**